### PR TITLE
Add forced mgmt route test case.

### PR DESCRIPTION
**Pull request by liuh-80**
Add forced mgmt route test case.

### Description of PR
Add forced mgmt route test case:
1. forced mgmt route add/del by port status.
1. forced mgmt route add/del will generate interfaces and route automatically.

##### Work item tracking
- Microsoft ADO: 24719238

### Type of change

- [ ] Bug fix
- [ ] Testbed and Framework(new/improvement)
- [x] Test case(new/improvement)

### Back port request
- [ ] 201911
- [ ] 202012
- [ ] 202205

#### What is the motivation for this PR?
Check and protect forced mgmt route feature: 
1. forced mgmt route add/del by port status.
1. forced mgmt route add/del will generate interfaces and route automatically.

#### How did you do it?
Add forced mgmt route test case:
1. forced mgmt route add/del by port status.
1. forced mgmt route add/del will generate interfaces and route automatically.

#### How did you verify/test it?
Pass all UT

### CODE REVIEW

File: tests/route/test_forced_mgmt_route.py [hunk 1]
```diff
@@ -0,0 +1,225 @@
...
+
+    yield
+
+    #  Restore config after test finish
+    restore_config(duthost, CONFIG_DB, CONFIG_DB_BACKUP)
+
+
+def get_interface_reload_timestamp(duthost):
+    res = duthost.command("sudo service interfaces-config status "
+                          "| grep 'Finished interfaces-config.service - Update interfaces configuration.'")
```
Comment (qiluo-msft) [inline tests/route/test_forced_mgmt_route.py:38]: >Finished [](http://example.com/codeflow?start=35&length=8)

Parsing human-readable output is fragile. Is it better to parse `systemctl show --no-pager interfaces-config` ?

File: tests/route/test_forced_mgmt_route.py [hunk 2]
```diff
@@ -0,0 +1,225 @@
...
+    logger.info("interfaces config timestamp {}".format(res["stdout_lines"]))
+
+    if len(res["stdout_lines"]) == 0:
+        return ""
+
+    return res["stdout_lines"][-1]
```
Comment (qiluo-msft) [inline tests/route/test_forced_mgmt_route.py:44]: >res["stdout_lines"][-1] [](http://example.com/codeflow?start=11&length=23)

If the output change a little bit, like a fixed non-time string at the end, will it fail or blindly succeed?

File: tests/route/test_forced_mgmt_route.py (hunk 1 above)
Comment (liuh-80) [reply tests/route/test_forced_mgmt_route.py]: Fixed

File: tests/route/test_forced_mgmt_route.py (hunk 2 above)
Comment (liuh-80) [reply tests/route/test_forced_mgmt_route.py]: Fixed, change to read ExecMainExitTimestamp from stdout.

**Review by qiluo-msft — APPROVED** [review]

**Comment by liuh-80** [comment]
This test case and https://github.com/sonic-net/sonic-mgmt/pull/11897 need cherry-pick to 202305/202311 because the feature backported to 202305.

**Pull request merged**